### PR TITLE
[CustomRuntime] migrate CustomRuntime into phi

### PR DESCRIPTION
**Pull request by ronny1996**
### PR types

Others
### PR changes

Others
### Describe

migrate CustomRuntime into phi

### CODE REVIEW

**Review by qili93 — APPROVED** [review]
LGTM

**Review by Aganlengzi — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by Shixiaowei02 — APPROVED** [review]
Approve namespace changes. Please contact @shangzhizhou @jiweibo additionally to confirm the inference availability. Thanks!

**Review by shangzhizhou — APPROVED** [review]
LGTM

File: paddle/phi/backends/callback_manager.h [hunk 1]
```diff
@@ -32,8 +32,7 @@
 
 #include "paddle/fluid/platform/enforce.h"
 
-namespace paddle {
```
Comment (jiweibo) [inline paddle/phi/backends/callback_manager.h:35]: callback_manager从fluid移动入phi内部，会引入头文件thread_pool.h   @chenwhql 请知悉

**Review by jiweibo — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**